File: Arrays/separate_the_digits/python/separate_digits.py
from typing import List

class Solution:
    def separateDigits(self, nums: List[int]) -> List[int]:
        results = []
        for i in nums:
            result = map(int, str(i))
            results.extend(result)
        return results

if __name__ == "__main__":
    test_cases = [
        [13, 25, 32],        # → [1, 3, 2, 5, 3, 2]
        [7, 1, 3, 9],        # → [7, 1, 3, 9], all single digits
        [100, 200],          # → [1, 0, 0, 2, 0, 0], zeros in digits
        [99, 77],            # → [9, 9, 7, 7]
        [1],                 # → [1], single element
    ]

    for nums in test_cases:
        print(f"Input:       {nums}")
        print(f"Output:      {Solution().separateDigits(nums)}")
        print("-" * 35)
    print(str(99).split())

File: Arrays/separate_the_digits/TS/separateDigits.ts
export {};

function separateDigits(nums: number[]): number[] {
  let results = [];
  for (const num of nums) {
    let result = Array.from(String(num)).map((i) => Number(i));
    results.push(...result);
  }
  return results;
}


const testCases: number[][] = [
    [13, 25, 32],
    [7, 1, 3, 9],
    [100, 200],
    [99, 77],
    [1],
];

for (const nums of testCases) {
    console.log(`Input:       ${JSON.stringify(nums)}`);
    console.log(`Output:      ${JSON.stringify(separateDigits(nums))}`);
    console.log("-".repeat(35));
}